deploy/wmagent-venv-deploy.sh

```diff
@@ -0,0 +1,760 @@
+                                  [-r <wmcore_source_repository>] [-b <wmcore_source_branch>] [-t <wmcore_tag>]
+                                  [-g <wmcore_config_repository>] [-d wmagent_path] [-l <service_list>] [-i <pypi_index>]
+                                  [-h <help>]
+
+      -s  <run_from_source>          Bool flag to setup run from source [Default: false]
+      -n  <no_venv_cleanup>          Bool flag to skip virtual environment space cleanup before deployment [Default: false - ALWAYS cleanup before deployment]
+      -v  <verbose_mode>             Bool flag to set verbose mode [Default: false]
+      -y  <assume yes>               Bool flag to assume 'Yes' to all deployment questions.
+      -r  <wmcore_source_repository> WMCore source repository [Default: git://github.com/dmwm/wmcore.git"]
+      -b  <wmcore_source_branch>     WMCore source branch [Default: master]
+      -t  <wmcore_tag>               WMCore tag to be used for this deployment [Default: None]
+      -d  <wmagent_path>             WMAgent virtual environment target path to be used for this deployment [Default: ./WMAgent.venv3]
+      -h <help>                      Provides help to the current script
+
+    # Example: Deploy WMAgent version 2.0.3rc1 from 'test' pypi index:
+    #          at destination /data/tmp/WMAgent.venv3/
+    # ./wmagent-venv-deploy.sh -i test -t 2.2.1 -d /data/tmp/WMAgent.venv3/
+
+    # Example: Same as above, but do not cleanup deployment area and reuse it from previous installtion - usefull for testing behaviour
+    #          of different versions during development or mix running from source and from pypi installed packages:
+    # ./wmagent-venv-deploy.sh -n -i test -t 2.2.1 -d /data/tmp/WMAgent.venv3/
+
+    # Example: Deploy WMAgent from source repository, use tag 2.2.1,
+    #          at destination /data/tmp/WMAgent.venv3/:
+    # ./wmagent-venv-deploy.sh -s -t 2.2.1 -d /data/tmp/WMAgent.venv3/
+
+    # Example: Same as above, but assume 'Yes' to all questions. To be used in order to chose the default flow and rely only
+    #          on the pameters set for configuring the deployment steps. This will avoid human intervention during deployment:
+    # ./wmagent-venv-deploy.sh -y -s -t 2.2.1 -d /data/tmp/WMAgent.venv3/
+
+    # DEPENDENCIES: All WMCore packages have OS or external libraries/packages dependencies, which are not having a pypi equivalent.
+    #               In the past those have been resolved through the set of *.spec files maintained at: https://github.com/cms-sw/cmsdist/tree/comp_gcc630
+    #               Currently they need to be resolved manually preveous to the Pypi based package installation.
+    #               Here follows the list of all direct (first level) dependencies per service generated from those spec files:
+
+    #               wmagent           : [python3, MariaDB, CouchDB]
+
+EOF
+}
+
+usage(){
+    help $*
+    exit 1
+}
+
+_realPath(){
```

Comment:
In the past I was simply using `realpath` unix command, but Valentin mentioned this was missing for MAC machines, so I developed this auxiliary function and extensively tested it. It has proved itself handy to prevent many mistakes.

Reply:
done